### PR TITLE
v1beta1 and Tinkerbell changes

### PR DESCRIPTION
**Pull request by panktishah26**
*Issue #, if available:*

*Description of changes:*
This branch is for testing all Tinkerbell related tasks. I will remove this branch once we have all the CAPI v1beta1 changes in the main branch.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/clustermanager/internal/deployments.go [hunk 1]
```diff
@@ -1,13 +1,10 @@
 package internal
 
-import "github.com/aws/eks-anywhere/pkg/types"
-
 // map where key = namespace and value is a capi deployment
 var CAPIDeployments = map[string][]string{
 	"capi-kubeadm-bootstrap-system":     {"capi-kubeadm-bootstrap-controller-manager"},
 	"capi-kubeadm-control-plane-system": {"capi-kubeadm-control-plane-controller-manager"},
 	"capi-system":                       {"capi-controller-manager"},
-	"capi-webhook-system":               {"capi-controller-manager", "capi-kubeadm-bootstrap-controller-manager", "capi-kubeadm-control-plane-controller-manager"},
```
Comment (ptrivedi) [inline pkg/clustermanager/internal/deployments.go:10]: Hmm, why is this removed?

File: pkg/clustermanager/internal/deployments.go [hunk 2]
```diff
@@ -16,23 +13,6 @@ var ExternalEtcdDeployments = map[string][]string{
...
-	"wh-kubeadm-control-plane-controller-manager.log":  {Name: "capi-kubeadm-control-plane-controller-manager", Namespace: "capi-webhook-system", Container: "manager"},
-	"cert-manager.log":                                 {Name: "cert-manager", Namespace: "cert-manager"},
-	"cert-manager-cainjector.log":                      {Name: "cert-manager-cainjector", Namespace: "cert-manager"},
-	"cert-manager-webhook.log":                         {Name: "cert-manager-webhook", Namespace: "cert-manager"},
-	"coredns.log":                                      {Name: "coredns", Namespace: "kube-system"},
-	"local-path-provisioner.log":                       {Name: "local-path-provisioner", Namespace: "local-path-storage"},
-	"capv-controller-manager.log":                      {Name: "capv-controller-manager", Namespace: "capv-system", Container: "manager"},
-	"wh-capv-controller-manager.log":                   {Name: "capv-controller-manager", Namespace: "capi-webhook-system", Container: "manager"},
-}
-
```
Comment (ptrivedi) [inline pkg/clustermanager/internal/deployments.go:35]: Also this?

File: pkg/executables/clusterctl_test.go [hunk 3]
```diff
@@ -95,7 +95,7 @@ func TestClusterctlInitInfrastructure(t *testing.T) {
 			wantExecArgs: []interface{}{
 				"init", "--core", core, "--bootstrap", bootstrap, "--control-plane", controlPlane, "--infrastructure", "aws:v0.6.4", "--config", test.OfType("string"),
 				"--bootstrap", etcdadmBootstrap, "--bootstrap", etcdadmController,
-				"--watching-namespace", constants.EksaSystemNamespace,
```
Comment (ptrivedi) [inline pkg/executables/clusterctl_test.go:98]: I think we want the watching-namespace to be there for v1alpha1 (for types other than DockerDatacenterKind

File: pkg/executables/clusterctl_test.go [hunk 4]
```diff
@@ -111,7 +111,7 @@ func TestClusterctlInitInfrastructure(t *testing.T) {
 			wantExecArgs: []interface{}{
 				"init", "--core", core, "--bootstrap", bootstrap, "--control-plane", controlPlane, "--infrastructure", "vsphere:v0.7.8", "--config", test.OfType("string"),
 				"--bootstrap", etcdadmBootstrap, "--bootstrap", etcdadmController,
-				"--watching-namespace", constants.EksaSystemNamespace, "--kubeconfig", "tmp/k.kubeconfig",
```
Comment (ptrivedi) [inline pkg/executables/clusterctl_test.go:114]: Same comment as above

File: pkg/features/features.go [hunk 5]
```diff
@@ -48,3 +49,10 @@ func TinkerbellProvider() Feature {
 		IsActive: globalFeatures.isActiveForEnvVar(TinkerbellProviderEnvVar),
 	}
 }
+
+func UseV1beta1BundleRelease() Feature {
+	return Feature{
+		Name:     "Use tags from v1beta1 bundle-releas.yaml",
```
Comment (ptrivedi) [inline pkg/features/features.go:55]: typo in bundle-release.yaml

File: pkg/providers/tinkerbell/config/template-cp.yaml [hunk 6]
```diff
@@ -49,11 +49,67 @@ spec:
         - DirAvailable--etc-kubernetes-manifests
         kubeletExtraArgs:
           provider-id: PROVIDER_ID
-    preKubeadmCommands:
-    - mkdir -p /etc/kubernetes/manifests && ctr images pull {{.kubeVipImage}}
-      && ctr run --rm --net-host {{.kubeVipImage}} vip /kube-vip manifest
-      pod --arp --interface $(ip -4 -j route list default | jq -r .[0].dev) --address {{.controlPlaneEndpointIp}} --controlplane
-      --leaderElection > /etc/kubernetes/manifests/kube-vip.yaml
```
Comment (ptrivedi) [inline pkg/providers/tinkerbell/config/template-cp.yaml:56]: Was it decided to remove this for now? Or do we want to figure the root cause out first?

File: pkg/clustermanager/internal/deployments.go (hunk 1 above)
Comment (mrajashree) [reply pkg/clustermanager/internal/deployments.go:10]: these changes are from the PR to start using v1beta1 bundle behind feature flag, commits from that PR are included in this one. In upstream v1.0.1 tags, webhook is deployed within the same pod and not a separate pod, hence there's no separate namespace/pod for webhooks.

File: pkg/clustermanager/internal/deployments.go (hunk 2 above)
Comment (mrajashree) [reply pkg/clustermanager/internal/deployments.go:35]: Reference: https://github.com/aws/eks-anywhere/pull/813#discussion_r767052994

File: pkg/executables/clusterctl_test.go (hunk 3 above)
Comment (mrajashree) [reply pkg/executables/clusterctl_test.go:98]: For reference: https://github.com/aws/eks-anywhere/issues/810

**Comment by mrajashree** [comment]
@panktishah26 [this PR ](https://github.com/aws/eks-anywhere/pull/813) is merged with a few more changes

**Comment by panktishah26** [comment]
Closing this PR as CAPI v1beta1 changes are already merged in the main branch.
